linera-core/src/client/mod.rs

```diff
@@ -896,12 +887,28 @@ where
         #[cfg(with_metrics)]
         let _latency = metrics::PREPARE_CHAIN_LATENCY.measure_latency();
 
-        // Verify that our local storage contains enough history compared to the
-        // expected block height. Otherwise, download the missing history from the
-        // network.
-        let next_block_height = self.next_block_height();
+        let mut info = self.sync_until(self.next_block_height()).await?;
+
+        if self.state().has_other_owners(&info.manager.ownership) {
+            // For chains with any owner other than ourselves, we could be missing recent
+            // certificates created by other owners. Further synchronize blocks from the network.
+            // This is a best-effort that depends on network conditions.
+            let nodes = self.validator_nodes().await?;
+            info = self.synchronize_chain_state(&nodes, self.chain_id).await?;
+        }
+        self.update_from_info(&info);
+        Ok(info)
+    }
+
+    // Verifies that our local storage contains enough history compared to the
+    // expected block height. Otherwise, downloads the missing history from the
+    // network.
+    async fn sync_until(
```

Comment:
nit: `synchronize_until` ?

Reply:
Done in [Review pt2](https://github.com/linera-io/linera-protocol/pull/2745/commits/1c2b92ece23d1d7623718d7ca723563505a14da8).